### PR TITLE
PD-4572 Python unit tests

### PR DESCRIPTION
**Pull request by forjayssake**
Test cover key components:

* contract tests validating requests and responses
* logging format tests 
* exception handling test

* Minor fixes to api schemas and additional documentation

### CODE REVIEW

**Review by davidelms77 — DISMISSED** [review]

**Review by terado — DISMISSED** [review]
pair worked and reviewed

File: project.code-workspace [hunk 1]
```diff
@@ -1,166 +0,0 @@
-{
```
Comment (dehe1) [inline project.code-workspace:1]: Is this meant to be removed?

Comment (forjayssake) [reply project.code-workspace]: Good catch, I've restored this

**Review by terado — APPROVED** [review]
all good code

**Review by dehe1 — APPROVED** [review]

**Pull request merged**